docroot/themes/custom/uids_base/assets/js/toggletip.js

```diff
@@ -0,0 +1,77 @@
+(function() {
+  // Get all the toggletip buttons
+  var toggletipTexts = document.querySelectorAll('[data-toggletip]');
+
+  // Iterate over them
+  Array.prototype.forEach.call(toggletipTexts, function(toggletipText) {
+    // Create the container element
+    var container = document.createElement('span');
+    container.setAttribute('class', 'toggletip-container');
+
+    // Put it before the original element in the DOM
+    toggletipText.parentNode.insertBefore(container, toggletipText);
+
+    // Create the button element
+    var toggletip = document.createElement('button');
+    toggletip.setAttribute('type', 'button');
+    toggletip.setAttribute('aria-label', 'more info');
```

Comment:
I think we should call it what it is. "featured image caption"

Reply:
actually, i think the label should be on the role status span if the intention is that it isn't read until clicked

Reply:
Counterpoint: we should leave it generic and/or potentially use another data attribute to populate the value rather than hardcoding it. To make it easier to apply to other things in the future.

Reply:
did this js get lifted from somewhere? i am wondering from a screen reader perspective if this is good UX. tab to the icon which says "more info" and that it is a button. click/enter creates a new element that is read off (status) but without a change in focus. esc to close or tab to close/move off.

Reply:
I had suggested https://codepen.io/heydon/pen/Vzwdpy as a starting point from https://inclusive-components.design/tooltips-toggletips/.

Reply:
The original codepen: https://codepen.io/heydon/pen/Vzwdpy. It's been updated a bit, but that's largely where it originates.
There was a post where this codepen originated as well, but I don't have that handy...
In general, from my understanding this UX pattern is preferable for 'toggletips', where screen readers see it on focus/click, and subsequent clicks essentially re-build it in order to re-read it if necessary. That option to re-read is where the lack of changing focus is supposed to be a benefit.

But even with that intention, the specifics of labels weren't always clear from my reading, and might need re-thinking.

Reply:
Thanks for pointing out the article. I understand the need to handle the mobile touch but it seems to take a step backwards from expected tabbing behavior.

<img width="51" alt="Screen Shot 2022-12-07 at 4 41 25 PM" src="https://user-images.githubusercontent.com/4663676/206312486-0b377bfa-aec3-4b4d-8de6-1fafb618e626.png">

As a keyboard user, I can tab through the elements and have the information revealed/read.
As a mouse user, I can hover on the elements to reveal the information.
As a mobile touch user, I can tap on the elements to reveal/close the information.
As a screen reader, if I am expected to click/enter something, I want to know what it is that I am clicking into so that I am not wasting time and not worried about getting trapped.

>potentially use another data attribute to populate the value rather than hardcoding it. To make it easier to apply to other things in the future.

I think that is a good idea.

Also as a visual user, I don't know this content exists unless I hover over/tab through the image and notice it. Should it be visible all the time and not fly in?

I am not saying we need to adopt or completely mimic it, but what can we take from Webform's implementation of https://atomiks.github.io/tippyjs/ that seems to address the user stories above.

<img width="466" alt="Screen Shot 2022-12-07 at 4 20 39 PM" src="https://user-images.githubusercontent.com/4663676/206310779-31271769-4d36-47b0-a6ac-5352dfd4a84a.png">

Reply:
So, I think it's first a question of if we want it as a tooltip or toggletip (as I've been differentiating it when discussing it, but have also seen it as popuptip as in https://www.nngroup.com/articles/tooltip-guidelines/). Both have a lot of overlap, and generally fill the same role, but function differently on how/when the info displays and how/when it goes away. I'm open to either, but I think that might help between what functionality is an established user pattern for this type that needs to be improved (eg button label), and what might be an argument for one type of pattern over another.

Re: fly-in or appearance, there was some discussion in midsprint demo about its level of prominence and wishing to tuck it away. Here it should appear on hover or focus over anything within the banner in order to 'reveal it' if it's possibly relevant, but maybe there's another way to lessen its prominence without hiding completely? Potentially positioning, a smaller (but still visible) indicator that becomes more prominent/clear upon interaction?

Reply:
Adding to previous, I think if we kept it a toggletip, we'd want to update the aria-label to something like "image caption" (better, make it dynamic so that it can be included as image caption for this, but otherwise have different descriptions) so that it is more accurate in the description, but still succinct as intended.

Reply:
Thanks for the library and style change. If this toggletip (new to me) behavior was decided during mid-sprint by the team, I am fine with shipping with this behavior to see how it is received by users. I would still like to see more context in the label wherever this is used.

Reply:
Nope! Still not decided. It's what was went with while working on it, and midsprint demo had feedback on the style, but it wasn't a solid decision on whether to use toggletip or tooltip. I think there's still room for discussion there, and whether we change it to tooltip, or roll with toggle and get feedback.

FWIW, toggletip as a concept is new to me too, I've just found various references to it since starting this work.

Reply:
> Re: fly-in or appearance, there was some discussion in midsprint demo about its level of prominence and wishing to tuck it away. Here it should appear on hover or focus over anything within the banner in order to 'reveal it' if it's possibly relevant, but maybe there's another way to lessen its prominence without hiding completely? Potentially positioning, a smaller (but still visible) indicator that becomes more prominent/clear upon interaction?

The thought was that by having it show on hover/focus, we would not have to display it on the image otherwise. This actually potentially allowed us to have the icon be more prominent (not need to use as much transparency and have a solid color icon) since it wouldn't be displaying all the time. When I suggested this, I was thinking of it more as something to try than as a decision per se.

Reply:
Way late to the party on this, but curious to know if an option with the caption below the featured image was considered before landing on a tooltip/toggle tip? Depending on the article and image, the caption text could hypothetically be critical to understanding the context of the image/photo used in an article.

Reply:
> Way late to the party on this, but curious to know if an option with the caption always visible below the featured image was considered before landing on a tooltip/toggle tip? Depending on the article and image, the caption text could hypothetically be critical to understanding the context of the image/photo used in the article and may need to be more readily visible.

So, we had brought it up as an option, but it we'd generally thought of that as the second option if the tool/toggletip didn't work out.

After thinking about it while working on it, I've also been wondering if there are some instances where we'd want it displayed directly on the page. One use-case that's been mentioned has been to list photo credits, and that to me seems like one instance where the credit should maybe not be tucked away. (In this case, might not be critical exactly, but good/courteous practice).

We could roll this out and later add the option to always display, but we might want some more discussion on those plans first.

Reply:
> @quamsta Yes, we considered that option as well. If you feel strongly about moving away from the toggle tip approach, now is a good time to chime in!

Definitely don't want to step on work that's already been done, but for things like image credits like @cory-skeers mentioned, they may be important to readily have available instead of putting them behind a user action. For the Mental Health site, I'm always putting image credits after the article text, but it'd be nice to have them up in the top area, where they'd make more sense imho. 

Quick-and-dirty mockup:

<img width="1181" alt="screenshot_2022-12-08" src="https://user-images.githubusercontent.com/472923/206560280-66f346ef-a9d3-4bff-b077-ed294e19248c.png">

(Would definitely need to be adjusted for mobile)

> So, we had brought it up as an option, but it we'd generally thought of that as the second option if the tool/toggletip didn't work out.
> 
> After thinking about it while working on it, I've also been wondering if there are some instances where we'd want it displayed directly on the page. One use-case that's been mentioned has been to list photo credits, and that to me seems like one instance where the credit should maybe not be tucked away. (In this case, might not be critical exactly, but good/courteous practice).
> 
> We could roll this out and later add the option to always display, but we might want some more discussion on those plans first.

Good thinking. An additional option may work. Would it be beneficial to ask content editors or people who work on Iowa Now about the preferred option here?

Reply:
Thanks for the additional feedback, @quamsta!

I think the plan will be to push forward with a version similar to your mockup here to get the caption field functional and on display on the page, and create a followup to gather feedback and potentially offer other options.

Reply:
That sounds awesome. Appreciate the flexibility here! Thank you for being receptive to some later-stage feedback.